src/api/tags/usePostTagsSuper.ts

```diff
@@ -0,0 +1,44 @@
+import { useState } from "react";
+import useApi from "../../hook/useApi";
+import { AxiosError, AxiosResponse } from "axios";
+import { Tag } from "../../types";
+import { setErrorDialog } from "../../data/error";
+
+type Props = {
+  bookInfoId: number;
+  addTag: (tag: Tag) => void;
+  setOpenTitleAndMessage: (
+    title: string,
+    message: string,
+    afterClose?: () => void,
+  ) => void;
+};
+
+export const usePostTagsSuper = ({
+  bookInfoId,
+  addTag,
+  setOpenTitleAndMessage,
+}: Props) => {
+  const [newTagName, setNewTagName] = useState("");
+
+  const { request } = useApi("post", "/tags/super", {
+    bookInfoId,
+    content: newTagName.replace(/ /g, "_"),
+  });
+
+  const onSuccess = (res: AxiosResponse) => {
+    const newTag = res.data as Tag;
```

Comment:
```suggestion
  const onSuccess = (res: AxiosResponse<Tag>) => {
    const newTag = res.data;
```
제네릭에 타입 지정해서 수정하겠습니다.

cf. 타입 단언, satisfies